catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -286,35 +329,69 @@ private String safeGetBody(Request req) {
     }
   }
 
-  private Map<String, Object> doCreateOrUpdate(Response response, Metacard metacard)
+  private Map<String, Object> doCreate(Response response, @Nullable Metacard metacard)
+      throws IngestException, SourceUnavailableException {
+    if (metacard == null) {
+      response.status(400);
+      return ImmutableMap.of(RESP_MSG, "Could not create metacard, no valid template specified");
+    }
+    if (metacard.getId() != null) {
+      LOGGER.debug(
+          "Cannot specify an ID [{}] when creating form/result metacard", metacard.getId());
+      response.status(400);
+      return ImmutableMap.of(RESP_MSG, "Could not create metacard, ID attribute not allowed");
```

Comment:
`ID attribute not allowed` is misleading; is that actually what is happening in the code?

Reply:
When creating a search form, it's important to note that the framework should be responsible for assigning the ID. If someone tries to include it in the input, we fail, because they are likely expecting a metacard with **that** ID to be created after the request when in reality it got a random ID from the framework. 

We could be more generic about it. Or we could use something similar to the log message above. I'm good with either.